how-to/workspace-platform-starter/client/src/framework/workspace/dock.ts

```diff
@@ -291,149 +294,148 @@ async function addEntryAsApp(
 				appId: entry.appId
 			}
 		}
-	});
+	};
 }
 
 /**
  * Add an entry to the dock as an action.
- * @param buttons The list of buttons to add to.
  * @param entry The entry details.
  * @param iconFolder The folder for icons.
  * @param colorSchemeMode The color scheme
+ * @returns The dock entry.
  */
 async function addEntryAsAction(
-	buttons: DockButton[],
 	entry: Omit<DockButtonAction, "id"> & { id?: string },
 	iconFolder: string,
 	colorSchemeMode: ColorSchemeMode
-): Promise<void> {
+): Promise<DockButton | undefined> {
 	if (!isStringValue(entry.tooltip)) {
 		logger.error("You must specify the tooltip for a DockButtonAction");
```

Comment:
Have we tried the unhappy path where there is no tooltip defined and undefined is returned?

Reply:
think so - we filter out the empty entries here

https://github.com/built-on-openfin/workspace-starter/blob/0f464c268062a12031c9ccb389ad23556da6ff7b/how-to/workspace-platform-starter/client/src/framework/workspace/dock.ts#L253